src/main/drivers/compass/compass_hmc5883l.c

```diff
@@ -259,50 +303,51 @@ static bool hmc5883lInit(void)
         LED1_TOGGLE;
     }
 
-    magGain[X] = fabsf(660.0f * HMC58X3_X_SELF_TEST_GAUSS * 2.0f * 10.0f / xyz_total[X]);
-    magGain[Y] = fabsf(660.0f * HMC58X3_Y_SELF_TEST_GAUSS * 2.0f * 10.0f / xyz_total[Y]);
-    magGain[Z] = fabsf(660.0f * HMC58X3_Z_SELF_TEST_GAUSS * 2.0f * 10.0f / xyz_total[Z]);
+    magdev->magGain[X] = ABS((int32_t)(660.0f * HMC58X3_X_SELF_TEST_GAUSS * 2.0f * 10.0f * 256.0f) / xyz_total[X]);
```

Comment:
BTW: is `ABS` necessary? Positive bias is added, negative subtracted, so result should be (2*) positive. the `fabs` looks like some old hack ... 

I would also prefer casting to `int`, there is no explicit need for 32 bits